src/base/matrix.rs

```diff
@@ -195,9 +195,32 @@ pub struct Matrix<T, R, C, S> {
     _phantoms: PhantomData<(T, R, C)>,
 }
 
-impl<T, R: Dim, C: Dim, S: fmt::Debug> fmt::Debug for Matrix<T, R, C, S> {
-    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
-        self.data.fmt(formatter)
+impl<T: fmt::Debug, R: Dim, C: Dim, S: RawStorage<T, R, C> + fmt::Debug> fmt::Debug
```

Comment:
Great job! Thanks for doing this. I've already started using some of your code in my local project!

I suggest not changing the approach to displaying matrices. This is code example:

```rust
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
        let (start, end) = match f.alternate() {
            true => {
                writeln!(f, "")?;
                ("[", "],\n ")
            },
            false => {
                write!(f, "[")?;
                ("[", "], ")
            },
        };

        for (i, row) in self.row_iter().enumerate() {
            if i > 0 {
                write!(f, "{end}")?;
            }
            write!(f, "{start}")?;
            for (j, element) in row.iter().enumerate() {
                if j > 0 {
                    write!(f, ", ")?;
                }
                element.fmt(f)?;
            }
        }

        match f.alternate() {
            true => writeln!(f, ""),
            false => write!(f, "]"),
        }
    }
```

Why this approach is better:
1. It allows you to copy data and directly paste it into the code, for example, for tests
2. It looks exactly like the matrices in the code

So, commas and square brackets have advantages over semicolons

PS: please note that the closed square bracket is missing in your implementation